### DISCUSSION TITLE
In question-answer blocks Copilot answers with unrelated StackOverflow links.

### DISCUSSION THREAD

**Discussion by lev-slinsen** [community/community#53189] [Copilot Conversations]
I've recently discovered the feature that allows you to ask Copilot questions in code comments with `q:`. And I started with really simple ones, in particular, related to Django framework. Instead of providing me with an answer, it provides links to StackOverflow that are completely unrelated to my questions, sometimes urls are just a combination of words from my own question.
Here's an example of what it suggests:
```
# q: How to get a list of objects filtered by name in Django framework?
# a: https://stackoverflow.com/a/10000000/1037948
```
And all other suggested solutions are just random unrelated SO threads.

**Comment by Akash1134** [comment]
Hi @lev-slinsen ,

Thanks for sharing your feedback, that means a lot. Regarding the issue you've highlighted above, as with most generative AI products, Copilot also makes its best guess based on the statistics and is subjected to making a few mistakes.

When GitHub Copilot synthesizes code suggestions based on available public data, it can also synthesize code that contains some undesirable patterns or public URLs. However, this is definitely an area of improvement and I've shared that with the concerned team. This is something we care a lot about at GitHub, and in recent years we’ve provided tools such as GitHub Actions, Dependabot, and CodeQL to open-source projects to help improve code quality. Of course, you should always use GitHub Copilot together with good testing and code review practices and security tools, as well as your own judgment.

I'd recommend reading these [FAQs](https://github.com/features/copilot#:~:text=Frequently%20asked%C2%A0questions) for more details about how Copilot suggests code and how you can leverage it at best. 🙂

**Reply by lev-slinsen** [reply]
Thank you. The point of this tool is to make coding easier and use less google requests, and this issue gets directly in the way. Hope it gets resolved soon. For now I'm going to revert to google/SO for any questions and leave Copilot only as a "code-completion" tool.
